### ISSUE TITLE
Delete `shrinkwrap`, then run `pnpm update foo`, removes all packages except `foo`

### ISSUE THREAD

**Issue by vjpr** [pnpm/pnpm#1200]
### Code to reproduce the issue:

```
npm init -y
pnpm i lodash underscore
rm shrinkwrap.yaml
pnpm update lodash
```

### Expected behavior:

Not sure exactly, but shouldn't remove packages that the `package.json` says should be installed.

### Actual behavior:

```
pnpm update lodash
Resolving: total 1, reused 1, downloaded 0, done
Packages: -1
-

dependencies:
- underscore 1.9.1
```

### Additional information:

 - `node -v` prints: 8.9
 - Windows, OS X, or Linux?: MacOS

**Comment by jiripospisil** [comment]
It's not even necessary to remove shrinkwrap.yaml to trigger some kind of weirdness. Consider this:

```
pnpm init -y
pnpm i lodash underscore
pnpm update lodash
```

```
pnpm update lodash

Already up-to-date

dependencies:
- lodash ^4.17.10
```

At this point lodash is not in package.json but it's still present in shrinkwrap.yaml and in node_modules

```
grep lodash package.json

grep lodash shrinkwrap.yaml
  lodash: 4.17.10
  /lodash/4.17.10:
  lodash: ^4.17.10
```

There are variants of the underlying bug. I first encountered this when I tried to upgrade the global PNPM (and a few other packages) and it actually removed itself - as in, the package was gone from my system. I had to install it again with NPM.

I encountered a similar issue a few minutes ago without giving any arguments to `pnpm update` (that's why I'm here). I'll try to extract a minimal repro from the project and post it here.

OS: MacOS
Node.js: v10.8.0
PNPM: 2.13.1

**Comment by jiripospisil** [comment]
Okay, this one actually physically removes the pino package from node_modules:

```
{
  "name": "foo",
  "private": true,
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
  },
  "author": "foo",
  "license": "UNLICENSED",
  "dependencies": {
    "config": "^1.30.0",
    "http-shutdown": "^1.2.0",
    "koa": "^2.5.0",
    "koa-body": "^2.6.0",
    "koa-helmet": "^4.0.0",
    "koa-router": "^7.4.0",
    "lodash": "^4.17.5",
    "pino": "^4.13.0"
  },
  "devDependencies": {
    "@types/koa": "^2.0.44",
    "@types/koa-router": "^7.0.27",
    "@types/lodash": "^4.14.104",
    "@types/node": "^10.1.3",
    "@types/pino": "^4.7.1",
    "tslint": "^5.9.1",
    "typescript": "^2.7.2"
  }
}
```

```
pnpm i
pnpm update
Already up-to-date
```

```
pnpm update pino
Already up-to-date

dependencies:
- pino ^4.13.0
```

```
pnpm update
Packages: -18
------------------

dependencies:
- pino 4.17.6
```

**Comment by zkochan** [comment]
I was refactoring update in v2.14 and seems like these issues are not reproducible anymore.
Try to update to `v2.14.4` of pnpm.

**Comment by zkochan** [comment]
the initial issue with removing `shrinkwrap.yaml` is still reproducible in v2.14.5

the issues described by @jiripospisil are fixed in the latest version (2.14.5)

**Comment by zkochan** [comment]
:ship: [2.17.5](https://github.com/pnpm/pnpm/releases/tag/v2.17.5)
